### PR TITLE
Final blank line

### PR DESCRIPTION
**Pull request by agwozdowski**

### CODE REVIEW

**Comment by nijikon** [comment]
Ping me when this is done, I will rebase and fix the commit message, so it will close the issue automatically.

File: lib/polishgeeks/dev-tools/command/final_blank_line.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        # Default paths which we want to exclude from analyse
+        DEFAULT_PATHS_TO_EXCLUDE = %w(
+          coverage tmp log vendor public
+          app/assets/images
+          app/assets/fonts
+        )
+
+        # Executes this command
+        # @return [Array] command output array with list of files
+        #   that don't have final blank line
```
Comment (mensfeld) [inline lib/polishgeeks/dev-tools/command/final_blank_line.rb]: @return not valid

File: lib/polishgeeks/dev-tools/command/final_blank_line.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          @output = []
+          @counter = 0
+
+          files_to_analyze.each do |file|
+            @counter += 1
+            @output << file if File.size(file) > 0 && IO.readlines(file).last[-1] != "\n"
+          end
+        end
+
+        # @return [Boolean] true if files have final blank line
```
Comment (mensfeld) [inline lib/polishgeeks/dev-tools/command/final_blank_line.rb]: true if all of analysed files have a final blank line (please update)

File: lib/polishgeeks/dev-tools/command/final_blank_line.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        def error_message
+          "Following files don't have final blank line: \n#{output.join("\n")}\n"
+        end
+
+        private
+
+        # @return [Array<String>] array with files to analyze
+        def files_to_analyze
+          # expression {*,.*} is needed because glob method don't take unix-like hidden files
+          files_from_path('**/{*,.*}') - excludes
```
Comment (mensfeld) [inline lib/polishgeeks/dev-tools/command/final_blank_line.rb]: please add app_root to point directly to apps files

File: lib/polishgeeks/dev-tools/config.rb [hunk 4]
```diff
@@ -23,6 +23,7 @@ def config
       attr_accessor :simplecov_threshold
       attr_accessor :rspec_files_structure_ignored
```
Comment (mensfeld) [inline lib/polishgeeks/dev-tools/config.rb]: could you turn it into %i( rspec_files_structure ) and append _ignored?

File: lib/polishgeeks/dev-tools/command/final_blank_line.rb [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        )
+
+        # Executes this command and set output and counter variables
+        def execute
+          @output = []
+          @counter = 0
+
+          files_to_analyze.each do |file|
+            @counter += 1
+            @output << sanitize(file) if File.size(file) > 0 && IO.readlines(file).last[-1] != "\n"
```
Comment (mensfeld) [inline lib/polishgeeks/dev-tools/command/final_blank_line.rb]: could you extract this: File.size(file) > 0 && IO.readlines(file).last[-1] != "\n" into a separate method?

**Pull request merged**